File: LC-1313/LC-1313.cpp
class Solution {
public:
    /**
     * Concepts: Naive Approach
     */
    vector<int> decompressRLElist(vector<int>& nums) {
        vector<int> RLElist;
        int numsLen = nums.size();
        
        if (numsLen == 0) return RLElist;
        for (int i = 0; i < numsLen - 1; i += 2) {
            int freq = nums[i], val = nums[i + 1];
            while (freq--) {
                RLElist.push_back(val);
            }
        }
        return RLElist;
    }
};

File: LC-1313/LC-1313.ts
/**
 * Concepts: Naive Approach
 */

function decompressRLElist(nums: number[]): number[] {
  let numsLen: number = nums.length
  if (numsLen === 0) return []

  let RLElist: number[] = []
  for (let i: number = 0; i < numsLen - 1; i += 2) {
    while (nums[i]--) {
      RLElist.push(nums[i + 1])
    }
  }
  return RLElist
}
